downgrade

```diff
@@ -384,5 +382,9 @@ cli() {
 
 if ((!LIB)); then
   set -e
+
+  TEXTDOMAINDIR=$(cd "$(basename "$0")"/../share/locale && pwd)
+  export TEXTDOMAIN=downgrade TEXTDOMAINDIR
```

Comment:
Also a nice touch. This approach removes the need for any `in` file and any extra configuration files as well. It is also a fair assumption for the locale directory to be located adjacent to where `downgrade` is installed, especially given that is the approach in our package-build.

Reply:
When running `downgrade vim`, I get the following error:

```
/usr/local/bin/downgrade: line 386: cd: downgrade/../share/locale: Not a directory
```

I guess this is because `basename "$0"` would evaluate to `downgrade`. We could change `basename "$0"` to `which "$0"`, but this would evaluate to `/usr/local/bin/downgrade` and it will not be possible to `cd` on this as this is a file and not a directory.

I propose the following change which I also tested locally to replace line 386:

```shell
TEXTDOMAINDIR=$(realpath $(dirname $(which "$0"))/../share/locale)
```

Reply:
That was silly, it should just be `dirname`, not `basename`. using `realpath` vs `&& pwd` doesn't matter, so I'm inclined to keep my style.

Reply:
Yes, tested just replacing `basename` with `dirname` and it works. Marking this as resolved.

Reply:
Sorry to open this conversation again, I came across another error. Changing `basename` to `dirname` works find for the installed file. However, this creates a problem for us during development. So let's say I made some changes to `downgrade` and want to test it from my local repo. Say I run `./downgrade vim`. Then I get the following error message:

```
./downgrade: line 386: cd: ./../share/locale: No such file or directory
```

Perhaps we add some error handling for `TEXTDOMAINDIR` to default to `/usr/share/local` if the output of the `cd` command is non-zero?

Reply:
Or alternatively:

```shell
if [ -f "$(dirname "$0")"/../share/locale ]; then 
    TEXTDOMAINDIR=$(cd "$(dirname "$0")"/../share/locale && pwd)
    export TEXTDOMAIN=downgrade TEXTDOMAINDIR
fi
```

Reply:
Ah yes, we should do something for in-repo `./downgrade` usage. I ran into this error with the test and fixed it by putting this under the `LIB` check. Can you tell I almost exclusively develop by tests?

Reply:
I think this is fixed now.